### PR TITLE
Mdmueller cython wrappers: Made Python wrappers for public Cython functions

### LINKED ISSUE #1323: Create python wrappers for all public, pure Cython functions
To allow for proper auto-complete and inspection.

### PR DESCRIPTION
**Pull request by sciunto**
Replaces #1432 

Closes #1323, https://github.com/scikit-image/scikit-image/issues/2047

### CODE REVIEW

**Comment by sciunto** [comment]
@scikit-image/core please, feel free to help to sort out the last issues. We need to merge this PR quickly.

File: skimage/draw/draw.py [hunk 1]
```diff
@@ -272,3 +272,373 @@ def set_color(img, coords, color, alpha=1):
...
+
+    Notes
+    -----
+    Andres method presents the advantage that concentric
+    circles create a disc whereas Bresenham can make holes. There
+    is also less distortions when Andres circles are rotated.
+    Bresenham method is also known as midpoint circle algorithm.
+    Anti-aliased circle generator is available with `circle_perimeter_aa`.
+
+    This function is a wrapper for Cython code.
```
Comment (ahojnnes) [inline skimage/draw/draw.py]: I don't think this comment is necessary...

File: skimage/draw/draw.py [hunk 2]
```diff
@@ -272,3 +272,373 @@ def set_color(img, coords, color, alpha=1):
...
+    Returns
+    -------
+    rr, cc : ndarray of int
+        Pixel coordinates of polygon.
+        May be used to directly index into an array, e.g.
+        ``img[rr, cc] = 1``.
+
+    Notes
+    -----
+    This function is a wrapper for Cython code.
```
Comment (ahojnnes) [inline skimage/draw/draw.py]: I don't think this comment is necessary...

File: skimage/draw/draw.py [hunk 3]
```diff
@@ -272,3 +272,373 @@ def set_color(img, coords, color, alpha=1):
...
+    rr, cc, val : (N,) ndarray (int, int, float)
+        Indices of pixels (`rr`, `cc`) and intensity values (`val`).
+        ``img[rr, cc] = val``.
+
+    Notes
+    -----
+    Wu's method draws anti-aliased circle. This implementation doesn't use
+    lookup table optimization.
+
+    This function is a wrapper for Cython code.
```
Comment (ahojnnes) [inline skimage/draw/draw.py]: I don't think this comment is necessary...

File: skimage/draw/draw.py [hunk 4]
```diff
@@ -272,3 +272,373 @@ def set_color(img, coords, color, alpha=1):
...
+    Returns
+    -------
+    rr, cc : (N,) ndarray of int
+        Indices of pixels that belong to the ellipse perimeter.
+        May be used to directly index into an array, e.g.
+        ``img[rr, cc] = 1``.
+
+    Notes
+    -----
+    This function is a wrapper for Cython code.
```
Comment (ahojnnes) [inline skimage/draw/draw.py]: I don't think this comment is necessary...

File: skimage/draw/draw.py [hunk 5]
```diff
@@ -272,3 +272,373 @@ def set_color(img, coords, color, alpha=1):
...
+        Indices of pixels that belong to the Bezier curve.
+        May be used to directly index into an array, e.g.
+        ``img[rr, cc] = 1``.
+
+    Notes
+    -----
+    The algorithm is the rational quadratic algorithm presented in
+    reference [1]_.
+
+    This function is a wrapper for Cython code.
```
Comment (ahojnnes) [inline skimage/draw/draw.py]: I don't think this comment is necessary...

File: skimage/feature/corner.py [hunk 6]
```diff
@@ -841,3 +842,125 @@ def corner_peaks(image, min_distance=1, threshold_abs=None, threshold_rel=0.1,
...
+        Window size.
+
+    Returns
+    -------
+    response : ndarray
+        Moravec response image.
+
+    Notes
+    -----
+    This function is a wrapper for Cython code.
```
Comment (ahojnnes) [inline skimage/feature/corner.py]: I don't think this comment is necessary...

File: skimage/measure/pnpoly.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        or anti-clockwise.  The first point may (but does not need to be)
+        duplicated.
+
+    See Also
+    --------
+    points_in_poly
+
+    Notes
+    -----
+    This function is a wrapper for Cython code.
```
Comment (ahojnnes) [inline skimage/measure/pnpoly.py]: I don't think this comment is necessary...

File: skimage/feature/corner.py [hunk 8]
```diff
@@ -841,3 +842,125 @@ def corner_peaks(image, min_distance=1, threshold_abs=None, threshold_rel=0.1,
...
+        calculation of the central moment.
+
+    Returns
+    -------
+    orientations : (N, 1) array
+        Orientations of corners in the range [-pi, pi].
+
+    Notes
+    -----
+    This function is a wrapper for Cython code.
```
Comment (ahojnnes) [inline skimage/feature/corner.py]: I don't think this comment is necessary...

File: skimage/segmentation/quickshift_wrapper.py [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    segment_mask : (width, height) ndarray
+        Integer mask indicating segment labels.
+
+    Notes
+    -----
+    The authors advocate to convert the image to Lab color space prior to
+    segmentation, though this is not strictly necessary. For this to work, the
+    image must be given in RGB format.
+
+    This function is a wrapper for Cython code.
```
Comment (ahojnnes) [inline skimage/segmentation/quickshift_wrapper.py]: I don't think this comment is necessary...

File: skimage/transform/hough_transform.py [hunk 10]
```diff
@@ -306,26 +181,115 @@ def hough_ellipse(img, threshold=4, accuracy=1, min_size=4, max_size=None):
...
-    >>> from skimage.transform import hough_ellipse
     >>> from skimage.draw import ellipse_perimeter
     >>> img = np.zeros((25, 25), dtype=np.uint8)
     >>> rr, cc = ellipse_perimeter(10, 10, 6, 8)
     >>> img[cc, rr] = 1
     >>> result = hough_ellipse(img, threshold=8)
-    >>> result.tolist()
-    [(10, 10.0, 10.0, 8.0, 6.0, 0.0)]
+    >>> result
+    [(10, 10.0, 8.0, 6.0, 0.0, 10.0)]
```
Comment (ahojnnes) [inline skimage/transform/hough_transform.py]: Why did this change?

File: skimage/transform/hough_transform.py [hunk 11]
```diff
@@ -306,26 +181,115 @@ def hough_ellipse(img, threshold=4, accuracy=1, min_size=4, max_size=None):
...
 
     Notes
     -----
     The accuracy must be chosen to produce a peak in the accumulator
     distribution. In other words, a flat accumulator distribution with low
     values may be caused by a too low bin size.
 
+    This function is a wrapper for Cython code.
```
Comment (ahojnnes) [inline skimage/transform/hough_transform.py]: I don't think this comment is necessary...

File: skimage/transform/hough_transform.py [hunk 12]
```diff
@@ -306,26 +181,115 @@ def hough_ellipse(img, threshold=4, accuracy=1, min_size=4, max_size=None):
...
+
+    Returns
+    -------
+    lines : list
+      List of lines identified, lines in format ((x0, y0), (x1, y0)),
+      indicating line start and end.
+
+    Notes
+    -----
+    This function is a wrapper for Cython code.
```
Comment (ahojnnes) [inline skimage/transform/hough_transform.py]: I don't think this comment is necessary...

**Comment by sciunto** [comment]
I think some tests do not pass...

**Comment by sciunto** [comment]
I fixed few additional regressions (thanks unittests).

**Comment by sciunto** [comment]
OK. I think I'm done fixing the tests.

File: skimage/measure/pnpoly.py [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+    For each ``(r, c)`` coordinate on a grid, i.e. ``(0, 0)``, ``(0, 1)`` etc.,
+    test whether that point lies inside a polygon.
+
+    Parameters
+    ----------
+    shape : tuple (M, N)
+        Shape of the grid.
+    verts : (V, 2) array
+        Specify the V vertices of the polygon, sorted either clockwise
+        or anti-clockwise.  The first point may (but does not need to be)
```
Comment (soupault) [inline skimage/measure/pnpoly.py]: Fix double whitespace.

File: skimage/transform/hough_transform.py [hunk 14]
```diff
@@ -125,17 +64,16 @@ def hough_line_peaks(hspace, angles, dists, min_distance=9, min_angle=10,
...
     distance_size = 2 * min_distance + 1
     angle_size = 2 * min_angle + 1
-    hspace_max = ndi.maximum_filter1d(hspace, size=distance_size, axis=0,
-                                      mode='constant', cval=0)
-    hspace_max = ndi.maximum_filter1d(hspace_max, size=angle_size, axis=1,
-                                      mode='constant', cval=0)
+    hspace_max = ndimage.maximum_filter1d(hspace, size=distance_size, axis=0,
+                                          mode='constant', cval=0)
+    hspace_max = ndimage.maximum_filter1d(hspace_max, size=angle_size, axis=1,
+                                          mode='constant', cval=0)
```
Comment (soupault) [inline skimage/transform/hough_transform.py]: I wonder why not just to use nd version of `maximum_filter`? (Not a blocker for this PR.)

File: skimage/transform/hough_transform.py [hunk 15]
```diff
@@ -257,43 +153,30 @@ def hough_circle(image, radius, normalize=True, full_output=False):
         Hough transform accumulator for each radius.
         R designates the larger radius if full_output is True.
         Otherwise, R = 0.
-
-    Examples
```
Comment (soupault) [inline skimage/transform/hough_transform.py]: I agree to remove this for now - is a very complicated example. Any plans to add something instead in the near future?

**Comment by soupault** [comment]
Thanks for pushing this, @sciunto! Few remarks from my side, then is good to go.

File: skimage/draw/draw.py [hunk 16]
```diff
@@ -272,3 +268,354 @@ def set_color(img, coords, color, alpha=1):
...
+    weight : double
+        Middle control point weight, it describes the line tension.
+    shape : tuple, optional
+        Image shape which is used to determine the maximum extent of output
+        pixel coordinates. This is useful for curves which exceed the image
+        size. If None, the full extent of the curve are used.
+
+    Returns
+    -------
+    rr, cc : (N,) ndarray of int
```
Comment (sciunto) [inline skimage/draw/draw.py]: If that's correct, it means that the API is wrong (see example below...). I vote to fix that elsewhere.

File: skimage/transform/hough_transform.py (hunk 15 above)
Comment (sciunto) [inline skimage/transform/hough_transform.py]: I even didn't figure out that was removed... hum.

**Comment by sciunto** [comment]
I made some changes. @soupault I do not disagree with the rest, but, if you don't mind, I would prefer to see new issues and this one merged. It's already thick. :)

**Comment by soupault** [comment]
@sciunto Sure.
You forgot to rename variables in `return`s. :)

**Comment by sciunto** [comment]
@soupault I opened #2319, so it's not blocking this PR and @amccaugh can work on his feature.

Thanks!

**Comment by soupault** [comment]
@sciunto this is blocking now -> https://github.com/scikit-image/scikit-image/pull/2303/commits/be94c6fde697036f0d7b5ca11fb96e45bd288c72#diff-17d59a5c1e414486337775a134925622L313 :). 0-indexing there as well.

**Comment by sciunto** [comment]
Thanks! I didn't see it.

**Comment by soupault** [comment]
@sciunto would you like to squash the commits, while you are here?

**Comment by sciunto** [comment]
I regrouped half of them. I kept some of them for the history.

**Comment by soupault** [comment]
This LGTM. @ahojnnes would you like to give it a quick look?

**Comment by ahojnnes** [comment]
LGTM.

**Comment by soupault** [comment]
Thanks @mdmueller, @sciunto! Great work, gentlemen!

**Pull request merged**